rust/worker/src/execution/operators/filter.rs

```diff
@@ -276,8 +226,9 @@ impl<'me> MetadataProvider<'me> {
                         PrimitiveOperator::GreaterThanOrEqual => Ok(reader.gte(key, kw).await?),
                         PrimitiveOperator::LessThan => Ok(reader.lt(key, kw).await?),
                         PrimitiveOperator::LessThanOrEqual => Ok(reader.lte(key, kw).await?),
-                        // Inequality filter is not supported at metadata provider level
-                        PrimitiveOperator::NotEqual => Err(FilterError::InvalidInput),
+                        PrimitiveOperator::NotEqual => unreachable!(
+                            "Inequality filter should be handled above the metadata provider level"
+                        ),
```

Comment:
Why unreachable rather than returning an error?  Last I checked, the compiler doesn't prove it's unreachable, it's just the same as a panic.

Reply:
It's not really an error, since the implementation should guarantee that this is never reached. It's effectively the same as `panic!`, but from my understanding it's a marker that reminds people this branch should not be reached (and if reached you should modify the code according to the texts here).